vm/runtime.ts

```diff
@@ -653,21 +654,25 @@ module J2ME {
   /** @const */ export var MIN_PRIORITY: number = 1;
   /** @const */ export var NORMAL_PRIORITY: number = 5;
 
+  /** @const */ export var ISOLATE_MIN_PRIORITY: number = 1;
+  /** @const */ export var ISOLATE_NORM_PRIORITY: number = 2;
+  /** @const */ export var ISOLATE_MAX_PRIORITY: number = 3;
+
   class PriorityQueue {
     private _top: number;
     private _queues: Context[][];
 
     constructor() {
-      this._top = MIN_PRIORITY;
+      this._top = MIN_PRIORITY + ISOLATE_MIN_PRIORITY;
       this._queues = [];
-      for (var i = MIN_PRIORITY; i <= MAX_PRIORITY; i++) {
+      for (var i = MIN_PRIORITY + ISOLATE_MIN_PRIORITY; i <= MAX_PRIORITY + ISOLATE_MAX_PRIORITY; i++) {
         this._queues[i] = [];
       }
     }
 
     enqueue(ctx: Context) {
-      var priority = ctx.getPriority();
-      release || assert(priority >= MIN_PRIORITY && priority <= MAX_PRIORITY,
+      var priority = ctx.getPriority() + ctx.runtime.priority;
```

Comment:
Can you try using ctx.runtime.isolate._priority instead? It shouldn't be much slower and should make it possible to change the Isolate priority at runtime (although we probably will never do it).